src/main/java/ch/njol/skript/expressions/ExprTime.java

```diff
@@ -76,56 +67,65 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 		switch (mode) {
 			case ADD:
 			case REMOVE:
-				return CollectionUtils.array(Timespan.class);
+				// allow time to avoid conversion to timespan, which causes all sorts of headaches
+				return CollectionUtils.array(Time.class, Timespan.class);
 			case SET:
 				return CollectionUtils.array(Time.class, Timeperiod.class);
-			case DELETE:
-			case REMOVE_ALL:
-			case RESET:
 			default:
 				return null;
 		}
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		final World[] worlds = getExpr().getArray(e);
-		int mod = 1;
-		switch (mode) {
-			case SET:
-				assert delta != null;
-				final int time = delta[0] instanceof Time ? ((Time) delta[0]).getTicks() : ((Timeperiod) delta[0]).start;
-				for (final World w : worlds) {
-					w.setTime(time);
-				}
-				break;
-			case REMOVE:
-				mod = -1;
-				//$FALL-THROUGH$
-			case ADD:
-				assert delta != null;
-				final Timespan ts = (Timespan) delta[0];
-				for (final World w : worlds) {
-					w.setTime(w.getTime() + mod * ts.getTicks());
-				}
-				break;
-			case DELETE:
-			case REMOVE_ALL:
-			case RESET:
-				assert false;
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (getExpr() == null || delta == null)
+			return;
+
+		Object time = delta[0];
+		if (time == null)
+			return;
+
+		World[] worlds = getExpr().getArray(event);
+
+		long ticks = 0;
+		if (time instanceof Time) {
+			if (mode != ChangeMode.SET) {
+				ticks = ((Time) time).getTicks() - TIME_TO_TIMESPAN_OFFSET; // allows for using "add 2:00" without going to new day
+			} else {
+				ticks = ((Time) time).getTicks();
+			}
+		} else if (time instanceof Timespan) {
+			ticks = ((Timespan) time).getAs(Timespan.TimePeriod.TICK);
+		} else if (time instanceof Timeperiod) {
+			ticks = ((Timeperiod) time).start;
+		}
+
+		for (World world : worlds) {
+			switch (mode) {
+				case ADD:
+					world.setTime(world.getTime() + ticks);
+					break;
+				case REMOVE:
+					world.setTime(world.getTime() - ticks);
+					break;
+				case SET:
+					world.setTime(ticks);
+					break;
+			}
```

Comment:
you could add the for loop to a helper method, and call the method in each of the `instanceof` checks, this avoids the need for the ticks variable (except for in `time instanceof Time`)

of course you dont have to do this, i just think it'll look a little neater

Reply:
good idea but i think this is easier to understand. if the team thinks your approach is better, i'll change it